openedx/features/enterprise_support/utils.py

```diff
@@ -300,18 +303,19 @@ def get_enterprise_learner_portals(request):
     # Prevent a circular import.
     from openedx.features.enterprise_support.api import enterprise_enabled
 
-    if enterprise_enabled():
+    user = request.user
+    # Only cache this if a learner is authenticated (AnonymousUser exists and should not be tracked)
+    if enterprise_enabled() and ENTERPRISE_HEADER_LINKS.is_enabled() and user and user.id:
```

Comment:
So the comment here is very useful to help me understand why `user and user.id` is here, but why does having the header link enabled need to be a condition here?

Reply:
Just adding it as a feature flag, if we need to revert this because of any reason it'll be easier to just flip the switch, we'll be able to fix forward then without rebasing and rolling back.